### PR TITLE
Add notification whether a goal is scored

### LINKED ISSUE #73: Notify robots that goal was scored
Via the message sent by supervisor

### PR DESCRIPTION
**Pull request by Adman**
* closes #73

Signed-off-by: Adrian Matejov <ado.matejov@gmail.com>

### CODE REVIEW

**Review by dbscoach — APPROVED** [review]
Looks good to me, have you run it yet?

**Comment by Adman** [comment]
@dbscoach yes I did ;)

File: controllers/rcj_soccer_referee_supervisor/referee/supervisor.py [hunk 1]
```diff
@@ -308,6 +308,9 @@ def _pack_packet(
         data.append(ball_translation[0])
         data.append(ball_translation[2])
 
+        # Add Notification if the goal is scored (True or False)
+        data.append(self.ball_reset_timer > 0)
```
Comment (mrshu) [inline controllers/rcj_soccer_referee_supervisor/referee/supervisor.py:312]: @Adman This is relatively nasty, this essentially mean the notification tells you whether we are in the time between when the goal gets scored and the new kickoff gets staged. I think calling it `is_goal` may be a bit dangerous in this regard (it is not set to true only in the "tick" when the goal gets scored).

Can you think of a name that's a bit more appropriate? I guess we should also explicitly describe this in the docs.

Thanks!

File: docs/docs/how_to.md [hunk 2]
```diff
@@ -67,6 +67,7 @@ class MyRobot:
             "x": unpacked[3 * N_ROBOTS],
             "y": unpacked[3 * N_ROBOTS + 1]
         }
+        data["is_goal"] = unpacked[3 * N_ROBOTS + 2]
```
Comment (mrshu) [inline docs/docs/how_to.md:70]: @Adman is there a serious reason why leaving `3 * N_ROBOTS` would be better than abstracting it out in here? Asking mostly out of curiosity :)

Thanks!

**Review by mrshu — APPROVED** [review]
Just a small comment, nothing major though :)

Thanks for putting this together!

File: controllers/rcj_soccer_referee_supervisor/referee/supervisor.py (hunk 1 above)
Comment (Adman) [reply controllers/rcj_soccer_referee_supervisor/referee/supervisor.py:312]: Yeah, we should describe entire "packet" string in docs.

would something like `waiting_for_kickoff` or something similar work?

File: docs/docs/how_to.md (hunk 2 above)
Comment (Adman) [reply docs/docs/how_to.md:70]: Yeah, I'll change the docs too make it more clear.

File: controllers/rcj_soccer_referee_supervisor/referee/supervisor.py (hunk 1 above)
Comment (mrshu) [reply controllers/rcj_soccer_referee_supervisor/referee/supervisor.py:312]: @Adman that does indeed sound very nice -- thanks!

File: docs/docs/how_to.md (hunk 2 above)
Comment (Adman) [reply docs/docs/how_to.md]: Done.

File: controllers/rcj_soccer_referee_supervisor/referee/supervisor.py (hunk 1 above)
Comment (Adman) [reply controllers/rcj_soccer_referee_supervisor/referee/supervisor.py:313]: Renamed.

**Pull request merged**